src/Evolution/Systems/ScalarAdvection/FiniteDifference/ReconstructWork.hpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim>
+class Direction;
+template <size_t Dim>
+class Element;
+template <size_t Dim>
+class ElementId;
+template <size_t Dim>
+class Mesh;
+template <typename TagsList>
+class Variables;
+namespace evolution::dg::subcell {
+class NeighborData;
+}  // namespace evolution::dg::subcell
+namespace gsl {
+template <typename>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace ScalarAdvection::fd {
+/*!
+ * \brief Reconstructs the scalar field \f$U\f$. All results are written into
+ * `vars_on_lower_face` and `vars_on_upper_face`.
+ *
+ * Need more detailed docs - direction, lower and upper face?
+ */
+template <size_t Dim, typename TagsList, typename Reconstructor>
+void reconstruct_work(
+    gsl::not_null<std::array<Variables<TagsList>, Dim>*> vars_on_lower_face,
+    gsl::not_null<std::array<Variables<TagsList>, Dim>*> vars_on_upper_face,
+    const Reconstructor& reconstruct,
+    const Variables<tmpl::list<Tags::U>> volume_vars,
+    const Element<Dim>& element,
+    const FixedHashMap<maximum_number_of_neighbors(Dim) + 1,
+                       std::pair<Direction<Dim>, ElementId<Dim>>,
+                       evolution::dg::subcell::NeighborData,
+                       boost::hash<std::pair<Direction<Dim>, ElementId<Dim>>>>
+        neighbor_data,
+    const Mesh<Dim>& subcell_mesh, const size_t ghost_zone_size) noexcept;
+
+/*!
+ * \brief Reconstructs the scalar field \f$U\f$ for a given direction. All
+ * results are written into `vars_on_face`.
+ *
+ * This is used on DG elements to reconstruct boundary correction on the element
```

Comment:
It doesn't reconstruction the boundary correction, it reconstructs `U` and then computes the neighbor's input into the boundary correction. The order of when you do reconstruction matters, so we need to be explicit. Why not use the dox from GRMHD? You could expand it, but that dox is technically correct